File: app/src/main/java/com/ytempest/wanandroid/di/module/http/event/NetEvent.java
package com.ytempest.wanandroid.di.module.http.event;

import androidx.annotation.NonNull;

/**
 * @author heqidu
 * @since 2020/7/6
 */
public class NetEvent {
    public long callId;
    public String url;
    public long callStartTime;
    public long callEndTime;
    public long dnsStartTime;
    public long dnsEndTime;
    public long connectStartTime;
    public long connectEndTime;
    public long secureConnectStart;
    public long secureConnectEnd;

    public long requestHeadersStart;
    public long requestHeadersEnd;
    public long requestBodyStart;
    public long requestBodyEnd;

    public long responseBodyStart;
    public long responseBodyEnd;
    public long responseHeadersStart;
    public long responseHeadersEnd;
    public long responseBodySize;

    public boolean requestSuccess;
    public String errorReason;

    @NonNull
    @Override
    public String toString() {
        return getClass().getSimpleName() +
                ":[" + "\n" +
                "callId: " + callId + ",\n" +
                "url: " + url + ",\n" +
                "callTime: " + (callEndTime - callStartTime) + ",\n" +
                "dnsTime: " + (dnsEndTime - dnsStartTime) + ",\n" +
                "connectTime: " + (connectEndTime - connectStartTime) + ",\n" +
                "secureTime: " + (secureConnectEnd - secureConnectStart) + ",\n" +
                "requestHeadersTime: " + (requestHeadersEnd - requestHeadersStart) + ",\t" +
                "requestBodyTime: " + (requestBodyEnd - requestBodyStart) + ",\n" +
                "responseHeadersTime: " + (responseHeadersEnd - responseHeadersStart) + ",\t" +
                "responseBodyTime: " + (responseBodyEnd - responseBodyStart) + ",\n" +
                "responseBodySize: " + responseBodySize + ",\n" +
                "requestResult: " + (requestSuccess ? "success" : "fail") + ",\n" +
                "errorReason: " + errorReason + ",];";
    }
}

File: app/src/main/java/com/ytempest/wanandroid/di/module/http/event/NetEvent.kt
package com.ytempest.wanandroid.di.module.http.event

/**
 * @author heqidu
 * @since 21-2-7
 */
class NetEvent {
    var callId: Long = 0
    var url: String? = null
    var callStartTime: Long = 0
    var callEndTime: Long = 0
    var dnsStartTime: Long = 0
    var dnsEndTime: Long = 0
    var connectStartTime: Long = 0
    var connectEndTime: Long = 0
    var secureConnectStart: Long = 0
    var secureConnectEnd: Long = 0

    var requestHeadersStart: Long = 0
    var requestHeadersEnd: Long = 0
    var requestBodyStart: Long = 0
    var requestBodyEnd: Long = 0

    var responseBodyStart: Long = 0
    var responseBodyEnd: Long = 0
    var responseHeadersStart: Long = 0
    var responseHeadersEnd: Long = 0
    var responseBodySize: Long = 0

    var requestSuccess = false
    var errorReason: String? = null

    override fun toString(): String {
        return """
            ${javaClass.simpleName}:[
            callId: $callId,
            url: $url,
            callTime: ${callEndTime - callStartTime},
            dnsTime: ${dnsEndTime - dnsStartTime},
            connectTime: ${connectEndTime - connectStartTime},
            secureTime: ${secureConnectEnd - secureConnectStart},
            requestHeadersTime: ${requestHeadersEnd - requestHeadersStart},	requestBodyTime: ${requestBodyEnd - requestBodyStart},
            responseHeadersTime: ${responseHeadersEnd - responseHeadersStart},	responseBodyTime: ${responseBodyEnd - responseBodyStart},
            responseBodySize: $responseBodySize,
            requestResult: ${if (requestSuccess) "success" else "fail"},
            errorReason: $errorReason,];
            """.trimIndent()
    }
}